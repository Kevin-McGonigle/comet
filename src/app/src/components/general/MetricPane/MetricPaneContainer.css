.metricPaneContainer {
    width: 510px;
}

.overviewContainer {
    display: flex;
    flex-grow: 1;
    margin-left: 10px;
    margin-right: 10px;
    align-items: stretch;
}