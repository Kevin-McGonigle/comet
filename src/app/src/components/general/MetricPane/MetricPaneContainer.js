import React from 'react';
import styles from './MetricPaneContainer.css';
import classnames from 'classnames';
import { Pane } from 'evergreen-ui';
import RadarGraph from '../Charts/RadarGraph';
import ForceDirectedGraph from '../Charts/ForceDirectedGraph';
import MetricPane from './MetricPane';

const cx = args => classnames(styles, args)

const metricInfo = {
    "Cyclomatic Complexity": {
        metricValue: 10,
        metricColour: "green"
    },
    "Logical Lines of Code": {
        metricValue: 10,
        metricColour: "orange"
    },
    "Lines of Comment": {
        metricValue: 10,
        metricColour: "green"
    },
    "Afferent Coupling": {
        metricValue: 10,
        metricColour: "green"
    },
    "Efferent Coupling": {
        metricValue: 10,
        metricColour: "red"
    },
    "Instability": {
        metricValue: 10,
        metricColour: "green"
    },
    "Method Cohesion": {
        metricValue: 10,
        metricColour: "red"
    },
    "Relational Cohesion": {
        metricValue: 10,
        metricColour: "green"
    },
    "Nesting Depth": {
        metricValue: 10,
        metricColour: "orange"
    }
}

const MetricPaneContainer = props => {
    return (
        <>
            <div className={cx("metricPaneContainer")}>
                <Pane
                    display="flex"
                    flexWrap="wrap"
                    background="tint2"
                    elevation={4}
                >
                    { Object.keys(metricInfo).map(metric => {
                        return (
                            <MetricPane 
                                key={metric}
                                metricName={metric}
                                metricValue={metricInfo[metric].metricValue}
                                metricColour={metricInfo[metric].metricColour}
                            />  
                        ) 
                    })}
                </Pane>      
            </div> 

            <div className={cx("overviewContainer")}>
                <Pane
                    display="flex"
                    flexDirection="column"
                    width="100%"
                    background="tint2"
                    elevation={4}
                >
                    <RadarGraph />
                </Pane>
            </div>
        </>
    )
};

export default MetricPaneContainer;
