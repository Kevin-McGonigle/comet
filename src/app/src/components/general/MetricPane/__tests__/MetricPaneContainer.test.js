import React from 'react';
import MetricPaneContainer from '../MetricPaneContainer';
import { shallow } from 'enzyme';
import "../../../../setupTests"

test('should render MetricPaneContainer as expected', () => { 
    const component = shallow(<MetricPaneContainer />);
    expect(component.html()).toMatchSnapshot();
});

