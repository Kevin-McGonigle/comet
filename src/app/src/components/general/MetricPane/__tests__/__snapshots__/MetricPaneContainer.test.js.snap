// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MetricPaneContainer as expected 1`] = `"<div class=\\"metricPaneContainer\\"><div class=\\"css-nil ðŸ“¦bs_kymczs ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-wrap_wrap ðŸ“¦box-szg_border-box\\"><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_q368mw ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Cyclomatic Complexity</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_1275103 ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Logical Lines of Code</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_q368mw ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Lines of Comment</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_q368mw ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Afferent Coupling</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_6xs4tx ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Efferent Coupling</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_q368mw ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Instability</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_6xs4tx ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Method Cohesion</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_q368mw ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Relational Cohesion</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div><div class=\\"css-nil ðŸ“¦b-top_1px-solid-E4E7EB ðŸ“¦b-rgt_1px-solid-E4E7EB ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦b-lft_1px-solid-E4E7EB ðŸ“¦bs_kymczs ðŸ“¦bg_1275103 ðŸ“¦b-btm-clr_black ðŸ“¦b-lft-clr_black ðŸ“¦b-rgt-clr_black ðŸ“¦b-top-clr_black ðŸ“¦flt_left ðŸ“¦w_120px ðŸ“¦h_120px ðŸ“¦mb_24px ðŸ“¦ml_24px ðŸ“¦mr_24px ðŸ“¦mt_24px ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box\\"><div class=\\"metricInfo\\"><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">Nesting Depth</span><span class=\\"ðŸ“¦color_black ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box\\">10</span></div></div></div></div><div class=\\"overviewContainer\\"><div class=\\"css-nil ðŸ“¦bs_kymczs ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦w_100prcnt ðŸ“¦box-szg_border-box\\"><canvas height=\\"150\\" width=\\"150\\"></canvas></div></div>"`;
